lms/djangoapps/shoppingcart/models.py

```diff
@@ -824,6 +824,69 @@ def generate_pdf_invoice(self, course, course_price, quantity, sale_price):
 
         return pdf_buffer
 
+    def __unicode__(self):
+        return "company: {} , internal reference: {} , customer reference number: {}".format(
+            self.company_name,
+            self.internal_reference,
+            self.customer_reference_number
+        )
+
+INVOICE_TRANSACTION_STATUSES = (
+
+    ('started', 'started'),
+    ('completed', 'completed'),
+    ('cancelled', 'cancelled'),
+    ('payment', 'payment'),
+    ('refund', 'refund'),
```

Comment:
@wedaly one thing @chrisndodge had mentioning which was compelling to me, was that these states will be easier to index and query rather than negative and positive values. Though these status do seem excessive and will have an added cost in maintaining.

What would canceled look like if we were to work strictly with amounts?